core/sail/api/src/test/java/org/eclipse/rdf4j/sail/helpers/NotifyingSailConnectionWrapperTest.java

```diff
@@ -85,24 +82,8 @@ public void tearDown() throws SailException {
 	@Test
 	public void testAddThenRemoveListener() throws SailException {
 		wrapper.addConnectionListener(listener);
-		addStatement("a");
-		assertThat(listener.getCount()).isEqualTo(1);
-		removeStatement("a");
-		assertThat(listener.getCount()).isEqualTo(0);
+		verify(delegate).addConnectionListener(listener);
 		wrapper.removeConnectionListener(listener);
-		addStatement("b");
-		assertThat(listener.getCount()).isEqualTo(0);
-	}
```

Comment:
Do some of the other tests check that you can add statements after the listener has been removed?

Reply:
I honestly don't know. I thought the whole test was badly written and simplified it so that I could at least integrate it. The point of this particular test is to check that connection listeners are properly added/remved, not to check if  statements can be added. It might be good to add more tests to this class to check for those things - part of reason I'm restructuring all this is also just to make it easier to find the right place to add a test.

Reply:
I have a bunch of tests for the notifying sail in the SHACL module. When I developed it I realized that there were a few edge cases that the notifying sail didn’t notify correctly. Things like adding and removing and re-adding the same statement in a single transaction.